index.js

```diff
@@ -1,74 +1,269 @@
+const ZERO = new decimal(0);
+const ONE = new decimal(1);
+const TEN = new decimal(10);
 
-const WEIS_IN_SATOSHI = WEIS_IN_ETH / SATOSHIS_IN_BTC;
+const SATOSHIS_TO_BTC_PRECISION = new decimal(8);
+const WEIS_TO_GWEI_PRECISION = new decimal(9);
+const GWEIS_TO_ETH_PRECISION = new decimal(9);
+const WEIS_TO_ETH_PRECISION = new decimal(18);
+
+const WEIS_IN_ETH = TEN.toPower(WEIS_TO_ETH_PRECISION);
+const WEIS_IN_GWEI = TEN.toPower(WEIS_TO_GWEI_PRECISION);
+const GWEIS_IN_ETH = TEN.toPower(GWEIS_TO_ETH_PRECISION);
+const SATOSHIS_IN_BTC = TEN.toPower(SATOSHIS_TO_BTC_PRECISION);
+
+const WEIS_IN_SATOSHI = WEIS_IN_ETH.dividedBy(SATOSHIS_IN_BTC);
+const MINIMUM_GWEI = ONE.dividedBy(WEIS_IN_GWEI);
+const MINIMUM_ETH = ONE.dividedBy(WEIS_IN_ETH);
+const MINIMUM_BTC = ONE.dividedBy(SATOSHIS_IN_BTC);
+
+/**
+ * Checks if the provided value is zero, returns true if it is, false otherwise.
+ * @param {decimal.Decimal.Value} value 
+ * @returns {boolean}
+ */
+const isZero = (value) => {
+    return value.comparedTo(ZERO) === 0;
+};
+
+/**
+ * Checks if the provided value is decimal, returns true if it is, false otherwise.
+ * @param {decimal.Decimal.Value} value 
+ * @returns 
+ */
+const isDecimal = (value) => {
+    return value.comparedTo(value.floor()) !== 0;
+};
+
+/**
+ * Checks if the provided value is negative, returns true if it is, false otherwise.
+ * @param {decimal.Decimal.Value} value 
+ * @returns {boolean}
+ */
+const isNegative = (value) => {
+    return value.comparedTo(ZERO) === -1;
+};
+
+/**
+ * This function helps to check if a base unit value is valid, 
+ * like satoshis or weis that are the base/smallest units of BTC and ETH respectively and cannot be negative or have decimals.
+ * 
+ * Throws an error if the amount in the smallest unit (satoshis or weis) is invalid.
+ * @param {decimal.Decimal.Value} baseUnitAmount 
+ * @returns {void}
+ * @throws {Error} If the amount in the base unit is invalid (negative or decimal)
+ */
+const checkValidBaseAmount = (baseUnitAmount) => {
+    // Return early if the value is 0
+    if(baseUnitAmount.comparedTo(ZERO) === 0) {
+        return;
+    }
+    if(isDecimal(baseUnitAmount) || isNegative(baseUnitAmount)) {
+        throw new Error('Amount in the smallest unit must be greater or equal than 0 and cannot have decimals.');
+    }
+};
+
+/**
+ * Checks if the value is negative. Throws error if it is.
+ * @param {decimal.Decimal.Value} value 
+ * @returns {void}
+ * @throws {Error} If the value is negative
+ * 
+ */
+const checkIsNegative = (value) => {
+    if(isNegative(value)) {
+        throw new Error('The value cannot be negative.');
+    }
+};
+
+/**
+ * Checks if the value is smaller than the minimum. Throws error if it is.
+ * @param {decimal.Decimal.Value} value 
+ * @param {decimal.Decimal.Value} minimum 
+ * @param {string} unit 
+ * @returns {void}
+ * @throws {Error} If the value is less than the minimum
+ */
+const checkMinimum = (value, minimum, unit = '') => {
```

Comment:
Same

Reply:
This function is not exported, so there's no way to test it directly without adding extra dependencies like `rewire` and `sinon`. It is being used and tested indirectly.